dbms/src/Storages/DeltaMerge/Delta/ColumnFileFlushTask.cpp

```diff
@@ -44,10 +45,17 @@ DeltaIndex::Updates ColumnFileFlushTask::prepare(WriteBatches & wbs)
         if (!task.block_data)
             continue;
 
-        IColumn::Permutation perm;
-        task.sorted = sortBlockByPk(getExtraHandleColumnDefine(context.is_common_handle), task.block_data, perm);
-        if (task.sorted)
-            delta_index_updates.emplace_back(task.deletes_offset, task.rows_offset, perm);
+        // Data sorting alters the sequence of data, requiring updates to the VersionChain,
+        // like `cloneWithUpdates` of DeltaIndex, yet it provides no benefit for VersionChain's data scanning.
+        // Meanwhile, the DeltaIndex inherently supports unordered delta data, such as unflushed or compacted data.
+        // Therefore, only perform data sorting when the VersionChain is disabled.
```

Comment:
So all the ColumnFiles in the delta layer is not guaranteed to be ordered by PK?